### ISSUE TITLE
10 -> 11: KeyError: 'toclose'

### ISSUE THREAD

**Issue by m3asmi** [OCA/OpenUpgrade#1787]
```
2019-04-01 10:54:13,022 24416 DEBUG db_migrateda OpenUpgrade: 2 rows affected
2019-04-01 10:54:13,045 24416 ERROR db_migrateda OpenUpgrade: fleet: error in migration script fleet/migrations/11.0.0.1/post-migration.py: 'toclose'
2019-04-01 10:54:13,045 24416 ERROR db_migrateda OpenUpgrade: 'toclose'
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/openupgradelib/openupgrade.py", line 1430, in wrapped_function
    if use_env2 else cr, version)
  File "/opt/odoo/OpenUpgrade/addons/fleet/migrations/11.0.0.1/post-migration.py", line 14, in migrate
    env['fleet.vehicle.log.contract'].scheduler_manage_contract_expiration()
  File "/opt/odoo/OpenUpgrade/addons/fleet/models/fleet_vehicle_cost.py", line 278, in scheduler_manage_contract_expiration
    expired_contracts.write({'state': 'expired'})
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_thread.py", line 291, in write
    track_self.message_track(tracked_fields, initial_values)
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_thread.py", line 494, in message_track
    tracking = self._message_track_get_changes(tracked_fields, initial_values)
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_thread.py", line 444, in _message_track_get_changes
    result[record.id] = record._message_track(tracked_fields, initial_values[record.id])
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_thread.py", line 467, in _message_track
    tracking = self.env['mail.tracking.value'].create_tracking_values(initial_value, new_value, col_name, col_info)
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_tracking_value.py", line 57, in create_tracking_values
    'old_value_char': initial_value and dict(col_info['selection'])[initial_value] or '',
KeyError: 'toclose'
2019-04-01 10:54:13,046 24416 ERROR db_migrateda odoo.modules.loading: Error executing post migration script for module fleet
: 'toclose'
2019-04-01 10:54:13,050 24416 WARNING db_migrateda odoo.modules.loading: Transient module states were reset
2019-04-01 10:54:13,051 24416 ERROR db_migrateda odoo.modules.registry: Failed to load registry
Traceback (most recent call last):
  File "/opt/odoo/OpenUpgrade/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/opt/odoo/OpenUpgrade/odoo/modules/loading.py", line 451, in load_modules
    force, status, report, loaded_modules, update_module, models_to_check, upg_registry)
  File "/opt/odoo/OpenUpgrade/odoo/modules/loading.py", line 337, in load_marked_modules
    upg_registry=upg_registry,
  File "/opt/odoo/OpenUpgrade/odoo/modules/loading.py", line 217, in load_module_graph
    migrations.migrate_module(package, 'post')
  File "/opt/odoo/OpenUpgrade/odoo/modules/migration.py", line 187, in migrate_module
    migrate(self.cr, installed_version)
  File "/usr/local/lib/python3.6/dist-packages/openupgradelib/openupgrade.py", line 1430, in wrapped_function
    if use_env2 else cr, version)
  File "/opt/odoo/OpenUpgrade/addons/fleet/migrations/11.0.0.1/post-migration.py", line 14, in migrate
    env['fleet.vehicle.log.contract'].scheduler_manage_contract_expiration()
  File "/opt/odoo/OpenUpgrade/addons/fleet/models/fleet_vehicle_cost.py", line 278, in scheduler_manage_contract_expiration
    expired_contracts.write({'state': 'expired'})
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_thread.py", line 291, in write
    track_self.message_track(tracked_fields, initial_values)
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_thread.py", line 494, in message_track
    tracking = self._message_track_get_changes(tracked_fields, initial_values)
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_thread.py", line 444, in _message_track_get_changes
    result[record.id] = record._message_track(tracked_fields, initial_values[record.id])
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_thread.py", line 467, in _message_track
    tracking = self.env['mail.tracking.value'].create_tracking_values(initial_value, new_value, col_name, col_info)
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_tracking_value.py", line 57, in create_tracking_values
    'old_value_char': initial_value and dict(col_info['selection'])[initial_value] or '',
KeyError: 'toclose'
2019-04-01 10:54:13,052 24416 CRITICAL db_migrateda odoo.service.server: Failed to initialize database `db_migrateda`.
Traceback (most recent call last):
  File "/opt/odoo/OpenUpgrade/odoo/service/server.py", line 945, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "/opt/odoo/OpenUpgrade/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/opt/odoo/OpenUpgrade/odoo/modules/loading.py", line 451, in load_modules
    force, status, report, loaded_modules, update_module, models_to_check, upg_registry)
  File "/opt/odoo/OpenUpgrade/odoo/modules/loading.py", line 337, in load_marked_modules
    upg_registry=upg_registry,
  File "/opt/odoo/OpenUpgrade/odoo/modules/loading.py", line 217, in load_module_graph
    migrations.migrate_module(package, 'post')
  File "/opt/odoo/OpenUpgrade/odoo/modules/migration.py", line 187, in migrate_module
    migrate(self.cr, installed_version)
  File "/usr/local/lib/python3.6/dist-packages/openupgradelib/openupgrade.py", line 1430, in wrapped_function
    if use_env2 else cr, version)
  File "/opt/odoo/OpenUpgrade/addons/fleet/migrations/11.0.0.1/post-migration.py", line 14, in migrate
    env['fleet.vehicle.log.contract'].scheduler_manage_contract_expiration()
  File "/opt/odoo/OpenUpgrade/addons/fleet/models/fleet_vehicle_cost.py", line 278, in scheduler_manage_contract_expiration
    expired_contracts.write({'state': 'expired'})
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_thread.py", line 291, in write
    track_self.message_track(tracked_fields, initial_values)
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_thread.py", line 494, in message_track
    tracking = self._message_track_get_changes(tracked_fields, initial_values)
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_thread.py", line 444, in _message_track_get_changes
    result[record.id] = record._message_track(tracked_fields, initial_values[record.id])
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_thread.py", line 467, in _message_track
    tracking = self.env['mail.tracking.value'].create_tracking_values(initial_value, new_value, col_name, col_info)
  File "/opt/odoo/OpenUpgrade/addons/mail/models/mail_tracking_value.py", line 57, in create_tracking_values
    'old_value_char': initial_value and dict(col_info['selection'])[initial_value] or '',
KeyError: 'toclose'
2019-04-01 10:54:13,053 24416 INFO db_migrateda odoo.service.server: Initiating shutdown
2019-04-01 10:54:13,053 24416 INFO db_migrateda odoo.service.server: Hit CTRL-C again or send a second signal to force the shutdown.
2019-04-01 10:54:13,053 24416 DEBUG db_migrateda odoo.service.server: current thread: <_MainThread(MainThread, started 140272817776448)>
2019-04-01 10:54:13,053 24416 DEBUG db_migrateda odoo.service.server: process <_MainThread(MainThread, started 140272817776448)> (False)
2019-04-01 10:54:13,053 24416 DEBUG db_migrateda odoo.service.server: --
```

**Comment by pedrobaeza** [comment]
Sorry, but this error has nothing to do with OpenUpgrade. Check possible custom modules that add the value `toclose` in old version, and why in new version is trying to change it.

**Comment by m3asmi** [comment]
**fleet 10.0** has state `toclose` and it has been renamed to `expired` in the **fleet 11.0**

**Comment by m3asmi** [comment]
look https://github.com/odoo/odoo/commit/bbcebcc12360eb7edca75ccb65180a954fcd6120

**Comment by pedrobaeza** [comment]
OK, I see. I was looking for field names, not for selection values. Reopening this. @cubells can you check this? Possibly a mapping is needed.

**Comment by m3asmi** [comment]
any doc how to create pre-migration.py ?

**Comment by pedrobaeza** [comment]
Well, not too much, apart from the existing migration scripts.

**Comment by m3asmi** [comment]
plz add this lines to `fleet/migrations/11.0.0.1/pre-migration.py`

```  
@openupgrade.migrate()
def migrate(env, version):
    openupgrade.copy_columns(env.cr, COLUMN_COPIES)
    openupgrade.map_values(
        env.cr, openupgrade.get_legacy_name('state'), 'state',
        [('toclose', 'expired')], table='fleet_vehicle_log_contract',
    )
```

**Comment by pedrobaeza** [comment]
OK, fixed slightly different with this commit: 9de9877b10842f048754ad43a4ad619d4df0c669
